Azure/sonic-buildimage
# [as7816-64x] Update to support DPB (Dynamic Port Breakout)

Signed-off-by: polly_hsu <polly_hsu@edge-core.com>



#### Why I did it
 Update to support as7816-64x DPB (Dynamic Port Breakout)

#### How I did it

- Download the SONiC master image with commit, 64dc08a372 dated 2022-04-05 from `SONiC Image Azure Pipelines`
- Update the file change of this PR accordingly

#### How to verify it
- Set the default t1 topology configuration
- Configure to breakout Ethernet0 from "1x100G[40G]" to "4x25G"
$ sudo config interface breakout Ethernet0 "4x25G" -f
- Check the SONiC DBP interface status is expected
$ show interfaces status

```
root@sonic:/etc/sonic# show platform summary
 
Platform: x86_64-accton_as7816_64x-r0
 HwSKU: Accton-AS7816-64X
 ASIC: broadcom
 ASIC Count: 1
 Serial Number: N/A
 Model Number: N/A
 Hardware Revision: N/A
 root@sonic:/etc/sonic# show ver
 
SONiC Software Version: SONiC.master.87430-64dc08a37
 Distribution: Debian 11.3
 Kernel: 5.10.0-8-2-amd64
 Build commit: 64dc08a37
 Build date: Tue Apr 5 22:16:03 UTC 2022
 Built by: AzDevOps@sonic-build-workers-001CJN
 
Platform: x86_64-accton_as7816_64x-r0
 HwSKU: Accton-AS7816-64X
 ASIC: broadcom
 ASIC Count: 1
 Serial Number: N/A
 Model Number: N/A
 Hardware Revision: N/A
 Uptime: 09:30:00 up 1:04, 1 user, load average: 2.83, 1.85, 1.53
 Date: Thu 07 Apr 2022 09:30:00

  root@sonic:/etc/sonic# sudo config interface breakout Ethernet0 "4x25G" -f
 Do you want to Breakout the port, continue? [y/N]: y
 Running Breakout Mode : 1x100G[40G]
 Target Breakout Mode : 4x25G
 Ports to be deleted :
 {
 "Ethernet0": "100000"
 }
 Ports to be added :
 {
 "Ethernet0": "25000",
 "Ethernet1": "25000",
 "Ethernet2": "25000",
 "Ethernet3": "25000"
 }
 After running Logic to limit the impact
 Final list of ports to be deleted :
 {
 "Ethernet0": "100000"
 }
 Final list of ports to be added :
 {
 "Ethernet0": "25000",
 "Ethernet1": "25000",
 "Ethernet2": "25000",
 "Ethernet3": "25000"
 }
 sonic_yang(6):Note: Below table(s) have no YANG models: KDUMP, SNMP, SNMP_COMMUNITY
 Below Config can not be verified, It may cause harm to the system
 {}
 Do you wish to Continue? [y/N]: y
 Breakout process got successfully completed.
 Please note loaded setting will be lost after system reboot. To preserve setting, run `config save`.
 root@sonic:/etc/sonic# config save
 Existing files will be overwritten, continue? [y/N]: y
 Running command: /usr/local/bin/sonic-cfggen -d --print-data > /etc/sonic/config_db.json
 root@sonic:/etc/sonic# show interfaces status
 Interface Lanes Speed MTU FEC Alias Vlan Oper Admin Type Asym PFC
 ----------- --------------- ------- ----- ----- --------------- ------ ------ ------- --------------- ----------
 Ethernet0 73 25G 9100 N/A Eth1/1(Port1) routed down down QSFP28 or later N/A
 Ethernet1 74 25G 9100 N/A Eth1/2(Port1) routed down down N/A N/A
 Ethernet2 75 25G 9100 N/A Eth1/3(Port1) routed down down N/A N/A
 Ethernet3 76 25G 9100 N/A Eth1/4(Port1) routed down down N/A N/A
```
#### Which release branch to backport (provide reason below if selected)




We have the customer is working on 202111

#### Description for the changelog

[as7816-64x] Update to support DPB (Dynamic Port Breakout)

#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
